lib/main_common.pm

```diff
@@ -1842,4 +1844,51 @@ sub load_sles4sap_tests {
     }
 }
 
+sub updates_is_applicable {
+    # we don't want live systems to run out of memory or virtual disk space.
+    # Applying updates on a live system would not be persistent anyway.
+    # Also, applying updates on BOOT_TO_SNAPSHOT is useless.
+    # Also, updates on INSTALLONLY do not match the meaning
+    return !get_var('INSTALLONLY') && !get_var('BOOT_TO_SNAPSHOT') && !get_var('DUALBOOT') && !get_var('UPGRADE') && !is_livesystem;
+}
+
+sub guiupdates_is_applicable {
+    return get_var("DESKTOP") =~ /gnome|kde|xfce|lxde/ && !check_var("FLAVOR", "Rescue-CD");
+}
+
+sub load_system_update_tests {
+    return if get_var('SYSTEM_UPDATED');
+    if (is_sle) {
+        if (x11tests_is_applicable() && gnomestep_is_applicable() && !is_staging) {
+            loadtest "update/updates_packagekit_gpk";
+        }
+        elsif (consolestep_is_applicable()) {
+            loadtest "update/zypper_up";
+        }
+    }
+    else {
+        return unless updates_is_applicable();
+        if (need_clear_repos) {
+            loadtest "update/zypper_clear_repos";
+        }
+        if (get_var('ZYPPER_ADD_REPOS')) {
+            loadtest "console/zypper_add_repos";
+        }
```

Comment:
@rwx788 do you think we can just extract the above two loadtest outside of the `is_sle/opensuse` branch? I think both should be applicable just based on the variables, e.g. `ZYPPER_ADD_REPOS`, and not need to be additionally depending on `is_sle/opensuse`

Reply:
This part was there long time ago, so I haven't touched it. As of now it's used only for openSUSE. Of course we can move control to variables instead of using variables we already have. But it won't help much as main.pms are spaghetti code which contains complex logic which cannot be always generalized.

Reply:
and this is why we are improving the code here. I was asking to simplify by just moving that part out of the `is_sle`-branch, nothing else, no new variables, no change in variables.